framework/platform/window.h

```diff
@@ -67,6 +67,14 @@ class Window
 	 */
 	virtual float get_dpi_factor() const = 0;
 
+	/**
+     * @return The scale factor for systems with heterogenous window and pixel coordinates
+     */
+	virtual float get_retina_factor() const
```

Comment:
Can implementation be kept inside the `.cpp` file?

Reply:
And typo `heterogeneous` (not `heterogenous`)

Reply:
will fix